pkg/generator/grafana/grafana.go

```diff
@@ -100,24 +90,66 @@ func (w *Grafana) Cleanup(ctx context.Context, jsonSpec *apiextensions.JSON, pre
 }
 
 func newClient(ctx context.Context, gen *genv1alpha1.Grafana, kclient client.Client, ns string) (*grafanaclient.GrafanaHTTPAPI, error) {
-	secret, err := resolvers.SecretKeyRef(ctx, kclient, resolvers.EmptyStoreKind, ns, &esmeta.SecretKeySelector{
-		Namespace: &ns,
-		Name:      gen.Spec.Auth.Token.Name,
-		Key:       gen.Spec.Auth.Token.Key,
-	})
-	if err != nil {
-		return nil, err
-	}
 	url := strings.TrimPrefix(gen.Spec.URL, "https://")
+
+	scheme := "https"
+	if strings.HasPrefix(url, "http://") {
+		scheme = "http"
+	}
+
+	// Trim the scheme from the URL
+	if strings.Contains(url, "://") {
+		_, url, _ = strings.Cut(url, "://")
+	}
+
 	cfg := &grafanaclient.TransportConfig{
 		Host:     url,
 		BasePath: "/api",
-		Schemes:  []string{"https"},
-		APIKey:   secret,
+		Schemes:  []string{scheme},
+	}
+
+	if err := setGrafanaClientCredentials(ctx, gen, kclient, ns, cfg); err != nil {
+		return nil, err
 	}
+
 	return grafanaclient.NewHTTPClientWithConfig(nil, cfg), nil
 }
 
+func setGrafanaClientCredentials(ctx context.Context, gen *genv1alpha1.Grafana, kclient client.Client, ns string, cfg *grafanaclient.TransportConfig) error {
+	// First try to use service account auth
+	if gen.Spec.Auth.Token != nil {
+		serviceAccountAPIKey, err := resolvers.SecretKeyRef(ctx, kclient, resolvers.EmptyStoreKind, ns, &esmeta.SecretKeySelector{
+			Namespace: &ns,
+			Name:      gen.Spec.Auth.Token.Name,
+			Key:       gen.Spec.Auth.Token.Key,
+		})
+		if err != nil {
+			return err
+		}
+
+		cfg.APIKey = serviceAccountAPIKey
+		return nil
+	}
+
+	// Next try to use basic auth
+	if gen.Spec.Auth.Basic != nil {
+		basicAuthPassword, err := resolvers.SecretKeyRef(ctx, kclient, resolvers.EmptyStoreKind, ns, &esmeta.SecretKeySelector{
+			Namespace: &ns,
+			Name:      gen.Spec.Auth.Basic.Password.Name,
+			Key:       gen.Spec.Auth.Basic.Password.Key,
+		})
+		if err != nil {
+			return err
+		}
+
+		cfg.BasicAuth = url.UserPassword(gen.Spec.Auth.Basic.Username, string(basicAuthPassword))
+		return nil
+	}
+
+	// No auth found, fail
+	return fmt.Errorf("no auth configuration found")
```

Comment:
Looking for feedback here. If another service handles auth (such as a service mesh or reverse proxy adding JWT or Header auth), then the client could successfully connect without needing any authentication.

Reply:
Is this a feature that you need want to support right now? Or is it more of a nice-to-have thing? 
If you don't need it, then we should leave it as you suggest and bail out with an error.

Reply:
I don't currently need it - just though that somebody else might find it useful. I'll leave the error for now, and if somebody else wants to integrate with Grafana without auth, they can file a separate PR.